sr25519-java/src/lib.rs

```diff
@@ -266,3 +266,45 @@ pub unsafe extern "system" fn Java_io_novasama_substrate_1sdk_1android_encrypt_S
     };
     try_or_throw_null!(jni_env, jni_env.byte_array_from_slice(secret.to_bytes().as_ref()))
 }
+
+/**
+ * Extract Public Key from a raw secret key (soft derivation result).
+ *
+ * @param secret: [32 bytes private key scalar | 32 bytes nonce]
+ * @return public key [32 bytes]
+ */
+#[no_mangle]
+pub unsafe extern "system" fn Java_io_novasama_substrate_1sdk_1android_encrypt_Sr25519_getPublicKeyFromSecret(
+    jni_env: JNIEnv,
+    _: JClass,
+    secret: jbyteArray,
+) -> jbyteArray {
+    let secret_vec = try_or_throw_null!(jni_env, jni_env.convert_byte_array(secret));
+
+    let secret_key = try_or_throw_null!(jni_env, create_secret(secret_vec.as_slice()));
+
+    let public_key = secret_key.to_public();
+
+    try_or_throw_null!(jni_env, jni_env.byte_array_from_slice(public_key.to_bytes().as_ref()))
+}
+
+/**
+ * Extract Public Key from a generation seed.
+ *
+ * @param seed: generation seed - input buffer of SR25519_SEED_SIZE bytes
+ * @return public key [32 bytes]
+ */
+#[no_mangle]
+pub unsafe extern "system" fn Java_io_novasama_substrate_1sdk_1android_encrypt_Sr25519_getPublicKeyFromSeed(
```

Comment:
Do we need that? Seems like we can just use `keypairFromSeed` to get both the private key and the public key.
We would like to keep the api surface minimal to avoid confusion